Rakefile

```diff
@@ -13,5 +13,12 @@ namespace :tests do
     t.pattern = 'spec/**/*_spec.rb'
```

Comment:
perhaps the issue is with this pattern that contain the integration tests as well. we can move all unit-tests under spec/unit/ and update this one not to pick them.